code/API_definitions/qos-profiles.yaml

```
@@ -0,0 +1,475 @@
+openapi: 3.0.3
+info:
+  title: QoS Profiles
+  description: |
+    The QoS Profiles API allows the discovery of Quality of Service (QoS) profiles and its characteristics.
+
+    # Introduction
+
+    QoS Profiles are used in conjunction with other Quality-On-Demand APIs, to offer the application developers the capability to request for stable latency (reduced jitter) or throughput for some specified application data flows between application clients (within a user device) and Application Servers (backend services). The developer has a pre-defined set of Quality of Service (QoS) profiles which they could choose from depending on their latency or throughput requirements. The QoS profiles are defined by the network operator and are mapped to the connectivity characteristics of the network.
+
+    # API functionality
+
+    The QoS Profiles API provides the following functionality:
+    - Discovery of all QoS profiles offered by the network operator
+    - Retrieve the characteristics of a specific QoS profile by name
+
+    How QoS profiles are mapped to connectivity characteristics are subject to agreements between the communication service provider and the API invoker. Within the CAMARA project, you can find a sample for such a mapping of QoS profiles. [CAMARA QoS Profiles Mapping Table (REFERENCE DRAFT)](https://github.com/camaraproject/QualityOnDemand/blob/main/documentation/API_documentation/QoSProfile_Mapping_Table.md)
+
+    # Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
+servers:
+  - url: "{apiRoot}/qos-profiles/v-wip"
```

```suggestion
  - url: "{apiRoot}/qos-profiles/v-{version}"
```

Can we parameterize this?  Perhaps this needs to be a separate issue?

---

There is no simple 1:1 relation between "version" and the version within basepath depending on the type of API version. See https://wiki.camaraproject.org/display/CAM/API+versioning for details.